kratos/utilities/math_utils.h

```diff
@@ -818,37 +818,40 @@ class MathUtils
  
 	return cross;
     }
-    
+
     /**
-     * Identical but it assumes that the output vector is given already the correct size
-     * @param a: First input vector
-     * @param b: Second input vector
-     * @return c: The resulting vector
+     * Performs the cross product of the two input vectors a,b
+     * a,b are assumed to be of size 3 (check is only performed on vector sizes in debug mode)
+     * @param a First input vector
+     * @param b Second input vector
+     * @param c The resulting vector
      */
     
-    static inline void CrossProduct(
-        array_1d<double, 3>& c, 
-        const array_1d<double, 3>& a, 
-        const array_1d<double, 3>& b
-        )
-    {
-        c[0] = a[1]*b[2] - a[2]*b[1];
-        c[1] = a[2]*b[0] - a[0]*b[2];
-        c[2] = a[0]*b[1] - a[1]*b[0];
-    }
-    
-    //general version    
-    template< class T1, class T2, class T3 >
+    template< class T1, class T2 , class T3>
     static inline void CrossProduct(T1& c, const T2& a, const T3& b ){
+#ifdef KRATOS_DEBUG
+        if (a.size() != 3 || b.size() != 3) 
```

Comment:
ok, but also check the size of c

Reply:
The problem is that c can be a Vector of zero size, then we need to resize, but can be an array_1d of size 2 for example, with no possibility of resize

Reply:
correct, so in debug throw an error if the size of c is not correct

Reply:
I added but some problems didn't work, that means I should correct all the vectors not initialized?

Reply:
i think you do need to have this check. If it throws an error it means it is wrong... also see my comments in #1100 about verifying if aliasing exists